source/developer_manual/design/dict_protocol.rst

```diff
@@ -0,0 +1,251 @@
+in the response line contain server's major and minor protocol versions (tab
+separated). The client also checks the protocol version and can decide to
+close the connection if versions do not match.
+
+Currently, dovec'ts client and server check that they support the same major
+version number. Minor version can be ignored.
+
+
+Other dict commands and their line format is described as follows.
+
+.. Note:: Prior to dovecot 2.3.17 none of the following commands included the
+          username for which the dict operation is performed. Instead, the
+          username from the initial handshake message was used.
+
+
+LOOKUP command
+^^^^^^^^^^^^^^
+
+Used to lookup a key value from dict.
+
+.. code-block:: none
+
+   C: L<key>TAB(<user>)
+
+
+ITERATE command
+^^^^^^^^^^^^^^^
+
+Used to iterate over a key path.
+
+.. code-block:: none
+
+   C: I<flags>TAB<max rows>TAB<path>TAB(<user>)
+
+The iteration flag is a integer with following options:
+
++----------------+--------+--------------------------------------------------+
+| flag           | value  | meaning                                          |
++================+========+==================================================+
+| RECURSE        | 0x01   | Recurse to all the sub-hierarchies               |
++----------------+--------+--------------------------------------------------+
+| SORT_BY_KEY    | 0x02   | Sort returned results by key                     |
++----------------+--------+--------------------------------------------------+
+| SORT_BY_VALUE  |  0x04  | Sort returned results by value                   |
++----------------+--------+--------------------------------------------------+
+| NO_VALUE       | 0x08   | Don't return values, only keys                   |
++----------------+--------+--------------------------------------------------+
+| EXACT_KEY      | 0x10   | Don't recurse at all. This is basically the same |
+|                |        | as LOOKUP comman dbut it will return all of the  |
+|                |        | rows instead of only the first one.              |
++----------------+--------+--------------------------------------------------+
+| FLAG_ASYNC     | 0x20   | Perform iteration asynchronously                 |
++----------------+--------+--------------------------------------------------+
+
+
+BEGIN command
+^^^^^^^^^^^^^
+
+Begins a dict transaction with the given ID.
+
+.. code-block:: none
+
+   C: B<id>TAB(<user>)
+
+Note that transactions commands don't have replies with exception of COMMIT.
+After a transaction is successfully started, transaction operations i.e.
+SET, UNSET, ATOMIC_INC, and TIMESTAMP can be used.
+
+COMMIT command
+^^^^^^^^^^^^^^
+
+Commit the transaction corresponding to the given ID.
+
+.. code-block:: none
+
+   C: C<transaction id>
+
+Note that the transaction is looked up from the ID so no username is required.
+
+COMMIT_ASYNC command
+^^^^^^^^^^^^^^^^^^^^
+
+Commit the transaction in asynchronous mode.
```

Comment:
I think lets describe this differently:

```
Obsolete command to commit the transaction. This hasn't been used by the dict-client since v2.2.24.


.. code-block:: none

   C: D<transaction id>
```

We should remove this entirely -> DOP-2677